tests/core/pyspec/eth2spec/test/validator/test_validator_unittest.py

```diff
@@ -0,0 +1,296 @@
+from eth2spec.test.context import spec_state_test, never_bls, with_all_phases
+from eth2spec.test.helpers.attestations import build_attestation_data
+from eth2spec.test.helpers.block import build_empty_block
+from eth2spec.test.helpers.deposits import prepare_state_and_deposit
+from eth2spec.test.helpers.keys import privkeys, pubkeys
+from eth2spec.test.helpers.state import next_epoch
+from eth2spec.utils import bls
+from eth2spec.utils.ssz.ssz_typing import Bitlist
+
+
+def run_is_candidate_block(spec, eth1_block, period_start, success):
+    result = spec.is_candidate_block(eth1_block, period_start)
+    if success:
+        assert result
+    else:
+        assert not result
+
+
+def get_min_new_period_epochs(spec):
+    return int(
+        spec.SECONDS_PER_ETH1_BLOCK * spec.ETH1_FOLLOW_DISTANCE * 2  # to seconds
+        / spec.SECONDS_PER_SLOT / spec.SLOTS_PER_EPOCH
+    )
+
+
+def get_mock_aggregate(spec):
+    return spec.Attestation(
+        data=spec.AttestationData(
+            slot=10,
+        )
+    )
+
+
+#
+# Becoming a validator
+#
+
+
+@with_all_phases
+@spec_state_test
+@never_bls
+def test_check_if_validator_active(spec, state):
+    active_validator_index = len(state.validators) - 1
+    assert spec.check_if_validator_active(state, active_validator_index)
+    new_validator_index = len(state.validators)
+    amount = spec.MAX_EFFECTIVE_BALANCE
+    deposit = prepare_state_and_deposit(spec, state, new_validator_index, amount, signed=True)
+    spec.process_deposit(state, deposit)
+    assert not spec.check_if_validator_active(state, new_validator_index)
+
+
+#
+# Validator assignments
+#
+
+
+@with_all_phases
+@spec_state_test
+@never_bls
+def test_get_committee_assignment_current_epoch(spec, state):
```

Comment:
I suggest we run a similar test for `next_epoch`
and then show a failure on `next_epoch + 1`